torchgeo/transforms/transforms.py

```diff
@@ -97,3 +98,57 @@ def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
             the unchanged input
         """
         return sample
+
+
+class AugmentationSequential(Module):  # type: ignore[misc]
+    """Wrapper around kornia AugmentationSequential to handle input dicts."""
+
+    def __init__(self, *args: Module, data_keys: List[str]) -> None:
+        """Initialize a new augmentation sequential instance.
+
+        Args:
+            *args: Sequence of kornia augmentations
+            data_keys: List of inputs to augment (e.g. ["image", "mask", "boxes"])
+        """
+        super().__init__()
+        self.data_keys = data_keys
+
+        keys = []
+        for key in data_keys:
+            if key == "image":
+                keys.append("input")
+            elif key == "boxes":
+                keys.append("bbox")
+            else:
+                keys.append(key)
+
+        self.augs = K.AugmentationSequential(*args, data_keys=keys)
+
+    def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
+        """Perform augmentations and update data dict.
+
+        Args:
+            sample: the input
+
+        Returns:
+            the augmented input
+        """
+        # Kornia augmentations require masks to be float
+        if "mask" in sample:
+            sample["mask"] = sample["mask"].to(torch.float)  # type:ignore[attr-defined]
+
+        inputs = [sample[k] for k in self.data_keys]
+        outputs_list: Union[Tensor, List[Tensor]] = self.augs(*inputs)
+        outputs_list = (
+            outputs_list if isinstance(outputs_list, list) else [outputs_list]
+        )
+        outputs: Dict[str, Tensor] = {
+            k: v for k, v in zip(self.data_keys, outputs_list)
+        }
+        sample.update(outputs)
+
+        # Convert masks to int
```

Comment:
Are we guaranteed that all masks will be integers? Is there a reason we need to do this?

Reply:
That's a good point. I don't believe I've seen a case where you would want masks to not be ints since Cross Entropy losses generally require the targets to be ints.

Possibly student/teacher distillation type experiments where the targets would be the soft probabilities? Even then the masks would only be computed in the forward pass and not provided by the dataloader.

Anyways, maybe a solution is to check the dtype of the mask prior to augmentation and just recast to that dtype post augmentation.

Reply:
Updated to cast masks after augmentation to whatever dtype they were prior to augmentation.